modules/utxo-lib/test/bitgo/signature.ts

```diff
@@ -0,0 +1,130 @@
+      signKeys.forEach((key) => {
+        txBuilder.sign(
+          vin,
+          Object.assign(key, { network }),
+          redeemScript,
+          getDefaultSigHash(network),
+          value,
+          witnessScript
+        );
+      });
+    });
+
+    return txBuilder.buildIncomplete();
+  }
+
+  function assertVerifySignatureEquals(
+    tx: Transaction,
+    value: boolean,
+    verificationSettings?: {
+      publicKey?: Buffer;
+      signatureIndex?: number;
+    }
+  ) {
+    tx.ins.forEach((input, i) => {
+      assert.strictEqual(
+        verifySignature(tx, i, outputAmount, verificationSettings),
+        value,
+        JSON.stringify(verificationSettings)
+      );
+    });
+  }
+
+  function getSignKeyCombinations(length: number): bip32.BIP32Interface[][] {
+    if (length === 0) {
+      return [];
+    }
+    if (length === 1) {
+      return fixtureKeys.map((k) => [k]);
+    }
+    return getSignKeyCombinations(length - 1)
+      .map((head) => fixtureKeys.filter((k) => !head.includes(k)).map((k) => [...head, k]))
+      .reduce((all, keys) => [...all, ...keys]);
+  }
+
+  function checkSignTransaction(signKeys: bip32.BIP32Interface[]) {
+    const tx = createSignedTransaction(fixtureKeys, signKeys);
+
+    // return true iff there are any valid signatures at all
+    assertVerifySignatureEquals(tx, signKeys.length > 0);
+
+    fixtureKeys.forEach((k) => {
+      // if publicKey is given, return true iff it is included in signKeys
+      assertVerifySignatureEquals(tx, signKeys.includes(k), { publicKey: k.publicKey });
+    });
+
+    // When transactions are signed, the signatures have the same order as the public keys in the outputScript.
+    const orderedSigningKeys = fixtureKeys.filter((fixtureKey) => signKeys.includes(fixtureKey));
+
+    [0, 1, 2].forEach((signatureIndex) => {
+      fixtureKeys.forEach((k) => {
+        // If no public key is given, return true iff any valid signature with given index exists.
+        assertVerifySignatureEquals(tx, signatureIndex < signKeys.length, {
+          signatureIndex,
+        });
+
+        // If publicKey and signatureIndex are provided only return if both match.
+        assertVerifySignatureEquals(tx, signatureIndex === orderedSigningKeys.indexOf(k), {
+          publicKey: k.publicKey,
+          signatureIndex,
+        });
+      });
+    });
+  }
+
+  describe(`verifySignature ${getNetworkName(network)} ${scriptType}`, function () {
+    [0, 1, 2, 3].forEach((length) => {
+      it(`returns true for selected keys (length=${length})`, function () {
+        getSignKeyCombinations(length).forEach((keys) => checkSignTransaction(keys));
+      });
+    });
+  });
+}
+
+describe('signature', function () {
+  getNetworkList()
+    .filter(isMainnet)
+    // The signing and verification methods are largely network-independent so let's focus on a
+    // single network to reduce test time.
+    // During development it might make sense to test all networks.
```

Comment:
I'm a bit confused by this comment since as I see it development is ongoing and the tests are here to support development. When would you envision it being worth testing every network? And would we do that by just editing this test to drop the `filter` statements above?

Reply:
My idea is to uncomment the `isBitcoin` filter below when touching code that could alter the behavior for other coins. 

When I run this for all coins locally, it takes ~20secs which more then doubles the overall test time!